### ISSUE TITLE
Is there any Chinese document?

### ISSUE THREAD

**Issue by dong-lufei** [rollup/rollup#4790]
https://rollupjs.org/guide/en/

**Comment by lukastaegert** [comment]
There is a very outdated one I think, but nobody maintains it. At this point, I would much rather look into machine translation because there are just no people to update translations.

**Comment by dong-lufei** [comment]
A good and convenient translation tool, Chinese users need to go through the firewall, the free translation function is less, and it is inconvenient to operate more. Wall tools are generally charged

**Comment by zhanglolo** [comment]
https://github.com/docschina/rollupjs.org but nobody maintain 😕

**Comment by ShenQingchuan** [comment]
> There is a very outdated one I think, but nobody maintains it. At this point, I would much rather look into machine translation because there are just no people to update translations.

@lukastaegert 
Hi, Luka. I'm the maintainer of Vite Chinese docs and one of the main translator in Vue docs translation.

I'm so excited that Rollup has a brand new doc site based on vitepress! Since Vue ecosystem's docs are almost built in vitepress, so I already got quite proficient in maintaining vitepress-based documents.

May I have the chance to join Rollup team and contribute for Chinese translation?

**Comment by lukastaegert** [comment]
That would be amazing! There is a growing demand for Chinese documentation, which I have sympathies for, but we cannot maintain it.
One problem is that there is really SO MUCH documentation. But I guess, starting with an AI generated documentation that is reviewed by a human would already go a long way.
How does the Vite documentation handle updates e.g. when new features are added? Do they update and release the English version first, and the Chinese translation is updated independently later when a translator finds time, or does it work differently?

**Comment by ShenQingchuan** [comment]
So glad to receive your reply! @lukastaegert 

For now, Vite docs translation between different languages are not under a very common practice, but in Chinese translation we figured out a way based on just Git commits!

Here's the process:
1. New features are committed to main branch, with some docs update
2. The bot/Github workflow of Chinese docs repo will pull commitS history directly from the main repo's "docs" directory (if the docs are an independent repo, then we'll setup a similar workflow to sync with that)

To be noticed that we do translation by following a very strict rule: "translation markdown files' line number must align with the original source"

This rule brings benefits for dealing with git conflicts. You're welcome to review some of merged Pull Requests in ［Vite Chinese docs](https://github.com/vitejs/docs-cn/pulls) to see more details.

3. Translator need to resolve these conflicts or do new translation for incremental docs as maintaining.


To achieve this stage, we're supposed to finish the translation stage first.

You're right! We can do it manually, or just use some machine translation helper like ChatGPT or deepL.

**Comment by ShenQingchuan** [comment]
@lukastaegert 

For more information to you:

- The workflow for Vue docs trasnslation sync: https://github.com/vuejs-translations/ryu-cho

- [ ] - Since we're using vitepress to build our docs project, and this project has some dependencies/imports from Rollup source codes, so we would like to keep an exactly same folder structure as the English "docs" project, maybe "docs-cn", but it seems not a good choice to put all these languages' translation under the repo's root dir. So, can we move the "docs" to "docs-i18n", put together to manage them uniformly. But it would be better do this movement earlier since our translation stage may cost a long time and accompanied by a lot of changes, in order to make the work less cumbersome.

**Comment by lukastaegert** [comment]
That looks amazing, thanks for sharing!

So if I understand you correctly, for Vue and Vite, the translated documentations are actually in separate repositories that are forks from the main repository? And technically, they are served from different servers?

For Rollup, documentation is now part of the main repository. The reason is to ensure that
* Every new feature is documented via the same pull request that creates the feature. That way, there is nearly nothing that is undocumented.
* It is easy to update the documentation together with the actual release. Basically one minute after release, the new documentation is online.
* The REPL that is included in the website can be used for local development

Documentation updates itself work by force-updating the branch `documentation-published` to whatever the documentation should be. So documentation can also be updated without a release.

I wonder if this would work with your approach? Technically, you could just fork the entire Rollup repository. Then it would also be easy to translate e.g. the REPL. Then we just need to figure out how to forward `cn.rollupjs.org` to the Chinese deployment. Which could probably be done by e.g. CloudFlare. But I would leave that to @Rich-Harris, who owns the domain.

**Comment by ShenQingchuan** [comment]
> That looks amazing, thanks for sharing!
> 
> So if I understand you correctly, for Vue and Vite, the translated documentations are actually in separate repositories that are forks from the main repository? And technically, they are served from different servers?
> 
> For Rollup, documentation is now part of the main repository. The reason is to ensure that
> 
> * Every new feature is documented via the same pull request that creates the feature. That way, there is nearly nothing that is undocumented.
> * It is easy to update the documentation together with the actual release. Basically one minute after release, the new documentation is online.
> * The REPL that is included in the website can be used for local development
> 
> Documentation updates itself work by force-updating the branch `documentation-published` to whatever the documentation should be. So documentation can also be updated without a release.
> 
> I wonder if this would work with your approach? Technically, you could just fork the entire Rollup repository. Then it would also be easy to translate e.g. the REPL. Then we just need to figure out how to forward `cn.rollupjs.org` to the Chinese deployment. Which could probably be done by e.g. CloudFlare. But I would leave that to @Rich-Harris, who owns the domain.

@lukastaegert 
Thanks for your kind suggestions! awesome!

My initial thought was to put the translated document and the original document in the same repository, so that they could be maintained in a unified manner in the future.

However, forking may be the best option. 

I can directly use Rollup to package the entire repository and modify the original document. So I will translate it on my own first. 

Perhaps there are other translators in the community who have already started the translation and may release it earlier than me. In the past, when we maintained the Vue & Vite Chinese documentation, we followed a principle: whoever completes 80% of the main content of the original document first, we will designate their repository as the official designated repository, and start some official promotion, call for participation, and assistance.

See you on the day when the translation is completed!

**Comment by lukastaegert** [comment]
> My initial thought was to put the translated document and the original document in the same repository

I admit that was also my initial thought, but a separate repository means you can just organize much more freely. If it brings an advantage, we can merge the Chinese translations into the main repository at any time, though.

**Comment by ShenQingchuan** [comment]
I've already setup a Netlify site: https://github.com/ShenQingchuan/rollup-docs-cn, which is behind https://github.com/ShenQingchuan/rollup-docs-cn.

I'd like to use my spare time and weekends to do the translation.

**Comment by ShenQingchuan** [comment]
> > My initial thought was to put the translated document and the original document in the same repository
> 
> 
> 
> I admit that was also my initial thought, but a separate repository means you can just organize much more freely. If it brings an advantage, we can merge the Chinese translations into the main repository at any time, though.

Hi，@lukastaegert

I'm happily to share with you our progress about Chinese docs translation, it seems like we're able to make it done in this weekend!

I'm wondering how to manage this repo after finish the initial translation.

For now, it's deployed on my own Netlify, how are things going to do with the transfering?

As you may refer, we Vue/Vite team setup a organization's "team", and only allow this team to manage the docs-(their lang) repo, which can give more liberty to the maintainers, and is able to gathering more native speakers to build this community!

I'd like to be part of this team, and keeping maintaining this docs repo.

**Comment by lukastaegert** [comment]
Great news! As a first step, I created https://github.com/orgs/rollup/teams/chinese-docs. I wonder if this is already enough to allow you to transfer the repository to the Rollup organization? Meanwhile, I will have a chat with @Rich-Harris about the deployment.

**Comment by ShenQingchuan** [comment]
> Great news! As a first step, I created https://github.com/orgs/rollup/teams/chinese-docs. I wonder if this is already enough to allow you to transfer the repository to the Rollup organization? Meanwhile, I will have a chat with @Rich-Harris about the deployment.

Glad to hear that.

I'm going to make the translation done first and make everything ready for transering.

Thanks for your kindness and all your support!

**Comment by ShenQingchuan** [comment]
@lukastaegert

**Comment by lukastaegert** [comment]
@ShenQingchuan oops, I somehow assumed that transferring the repository what give you all necessary privileges. I gave you back admin privileges to your repo now 😉. And you are also now the maintainer of https://github.com/orgs/rollup/teams/chinese-docs so you can add new members here or change the name if you want to.

**Comment by lukastaegert** [comment]
I also added the repo to the team https://github.com/orgs/rollup/teams/chinese-docs/repositories

**Comment by ShenQingchuan** [comment]
> I also added the repo to the team https://github.com/orgs/rollup/teams/chinese-docs/repositories Let me know if you need anything else. If you can give me a deployment url, I can ask Rich to set up `cn.rollupjs.org`.

@lukastaegert  For now, I've posted it on the repo's sidebar link area, it's

https://rollup-docs-cn.netlify.app

Thanks for your supports!

**Comment by ShenQingchuan** [comment]
> 

@lukastaegert 

it seems like that I can't invite someone else outside Rollup org, I'm only able to invite someone who's already inside Rollup org. 😂

**Comment by lukastaegert** [comment]
Ooops, I see. So I guess you would need to be a maintainer to do that. As I do not want to step on the toes of any other maintainer in the Rollup organization for now, would it be ok for you to just send me the Github names of anyone you would want to add and I send them the invites?

**Comment by lukastaegert** [comment]
Ooops, I see. So I guess you would need to be a maintainer to do that. As I do not want to step on the toes of any other maintainer in the Rollup organization for now, would it be ok for you to just send me the Github names of anyone you would want to add and I send them the invites?

**Comment by ShenQingchuan** [comment]
> Ooops, I see. So I guess you would need to be a maintainer to do that. As I do not want to step on the toes of any other maintainer in the Rollup organization for now, would it be ok for you to just send me the Github names of anyone you would want to add and I send them the invites?

Sure! I understand your consideration for controlling the size of the organization. Appreciate your support and kindness!

I'd like to invite these workmate to join Rollup.js Chinese docs maintainer team:

- @SnowingFox  https://github.com/SnowingFox
- @waynzh https://github.com/waynzh
- @iDestin https://github.com/iDestin

There're a lot of translation work done by their PR, and they have shown patience, cooperation and dedication.
So I sincerely invite them to join the team.

Thanks!
@lukastaegert

**Comment by lukastaegert** [comment]
Done!

**Comment by iDestin** [comment]
I'm very excited to receive this news, thank you very much @ShenQingchuan for the recognition and @lukastaegert for the invitation, I will do my best to do what I can to help make Chinese documentation better!

**Comment by ShenQingchuan** [comment]
@lukastaegert 

Hi Lukas, since we've already done the first translation and the Chinese docs site is already online.
So I think maybe we can close this issue for now.

And maybe we can talk about the **official deployment** later?

**Comment by lukastaegert** [comment]
Sure thing! I contacted Rich to look into setting up cn.rollupjs.org, will get back to you once this is done.

**Comment by lukastaegert** [comment]
@ShenQingchuan Apparently, Rich already configured `cn.rollupjs.org` to forward to https://rollup-docs-cn.netlify.app, but you will need to configure Netlify to support the new public name.

**Comment by ShenQingchuan** [comment]
> @ShenQingchuan Apparently, Rich already configured `cn.rollupjs.org` to forward to https://rollup-docs-cn.netlify.app, but you will need to configure Netlify to support the new public name.

@lukastaegert 

I apologize for my lack of experience in configuring on Netlify. Could you kindly guide me on where to adjust my configuration? Your help is much appreciated. Thank you.

**Comment by lukastaegert** [comment]
Ah I see, I guess you are stuck here: 
<img width="379" alt="Bildschirm­foto 2023-03-19 um 19 33 56" src="https://user-images.githubusercontent.com/12949268/226199185-7205c6db-aad7-4ec4-b4ea-a142b813a300.png">
Then we probably need to do it through Rich's account, I will take care

**Comment by TLovers** [comment]
It seems that cn.rollupjs.org cannot be accessed

**Comment by lukastaegert** [comment]
Sorry, there have been some delays. Still on it.

**Comment by lukastaegert** [comment]
cn.rollupjs.org is now set up correctly! We just need to update an merge #4914 now.
